spec/index.bs

```diff
@@ -6796,8 +6806,198 @@ enum GPUVertexFormat {
 };
 </script>
 
-The multi-component formats specify the number of components after "x".
-As such, {{GPUVertexFormat/"sint32x3"}} denotes a 3-component vector of `i32` values in the shader.
+<table class="data">
+  <thead>
+    <tr>
+      <th>Vertex format
+      <th>Data type
+      <th>Components
+      <th>Byte size
+      <th>[=natural WGSL type=]
```

Comment:
The shader author has freedom over the component type's bit width.  So we should try to avoid biasing toward 32-bit types.

It seems a bit wobbly editorially to define a "natural" size type that then can be so heavily modified: component count and component bitwidth

Ideally WGSL would have definitions for the type kinds we want:  signed integer scalar, unsigned integer scalar, floating point scalar (being f16 or f32 for now).  Then make this last column "Shader component kind".

Let's not block on this now. We can do a followup which coordinates with a WGSL change.

Reply:
Thanks for the feedback! I know you suggested just landing and iterating later, but I've taken another pass at this section to try and better explain what mappings are possible, adding a table to show which WGSL scalar types can be used with which vertex data types and an example section showing all the possible mappings for a single format. Please take another look and let me know if any of it seems off?